pkg/descheduler/policyconfig_test.go

```diff
@@ -0,0 +1,444 @@
+	"sigs.k8s.io/descheduler/pkg/utils"
+	"sigs.k8s.io/descheduler/test"
+)
+
+func TestV1alpha1ToV1alpha2(t *testing.T) {
+	defaultEvictorPluginConfig := api.PluginConfig{
+		Name: defaultevictor.PluginName,
+		Args: &defaultevictor.DefaultEvictorArgs{
+			PriorityThreshold: &api.PriorityThreshold{
+				Value: utilpointer.Int32(utils.SystemCriticalPriority),
+			},
+		},
+	}
+	defaultEvictorPluginSet := api.PluginSet{
+		Enabled: []string{defaultevictor.PluginName},
+	}
+	type testCase struct {
+		description string
+		policy      *v1alpha1.DeschedulerPolicy
+		err         error
+		result      *api.DeschedulerPolicy
+	}
+	testCases := []testCase{
+		{
+			description: "RemoveFailedPods enabled, LowNodeUtilization disabled strategies to profile",
+			policy: &v1alpha1.DeschedulerPolicy{
+				Strategies: v1alpha1.StrategyList{
+					removeduplicates.PluginName: v1alpha1.DeschedulerStrategy{
+						Enabled: true,
+						Params: &v1alpha1.StrategyParameters{
+							Namespaces: &v1alpha1.Namespaces{
+								Exclude: []string{
+									"test2",
+								},
+							},
+						},
+					},
+					nodeutilization.LowNodeUtilizationPluginName: v1alpha1.DeschedulerStrategy{
+						Enabled: false,
+						Params: &v1alpha1.StrategyParameters{
+							NodeResourceUtilizationThresholds: &v1alpha1.NodeResourceUtilizationThresholds{
+								Thresholds: v1alpha1.ResourceThresholds{
+									"cpu":    v1alpha1.Percentage(20),
+									"memory": v1alpha1.Percentage(20),
+									"pods":   v1alpha1.Percentage(20),
+								},
+								TargetThresholds: v1alpha1.ResourceThresholds{
+									"cpu":    v1alpha1.Percentage(50),
+									"memory": v1alpha1.Percentage(50),
+									"pods":   v1alpha1.Percentage(50),
+								},
+							},
+						},
+					},
+				},
+			},
+			result: &api.DeschedulerPolicy{
+				Profiles: []api.Profile{
+					{
+						Name: fmt.Sprintf("strategy-%s-profile", removeduplicates.PluginName),
+						PluginConfigs: []api.PluginConfig{
+							defaultEvictorPluginConfig,
+							{
+								Name: removeduplicates.PluginName,
+								Args: &removeduplicates.RemoveDuplicatesArgs{
+									Namespaces: &api.Namespaces{
+										Exclude: []string{
+											"test2",
+										},
+									},
+								},
+							},
+						},
+						Plugins: api.Plugins{
+							Balance: api.PluginSet{
+								Enabled: []string{removeduplicates.PluginName},
+							},
+							Evict: defaultEvictorPluginSet,
+						},
+					},
+					// Disabled strategy is not generating internal plugin since it will not be used internally
+					// {
+					// 	Name: nodeutilization.LowNodeUtilizationPluginName,
+					// 	PluginConfigs: []api.PluginConfig{
+					// 		{
+					// 			Name: nodeutilization.LowNodeUtilizationPluginName,
+					// 			Args: &nodeutilization.LowNodeUtilizationArgs{
+					// 				Thresholds: api.ResourceThresholds{
+					// 					"cpu":    api.Percentage(20),
+					// [...]
+					// [...]
+					// },
```

Comment:
Related to https://github.com/kubernetes-sigs/descheduler/pull/999/files#r1019286039